x-pack/legacy/plugins/ml/public/application/datavisualizer/index_based/page.tsx

```diff
@@ -585,84 +587,96 @@ export const Page: FC = () => {
-                />
-                <EuiSpacer size="m" />
-                <EuiFlexGroup gutterSize="m">
-                  <EuiFlexItem>
-                    {totalMetricFieldCount > 0 && (
-                      <Fragment>
-                        <FieldsPanel
-                          title={i18n.translate('xpack.ml.datavisualizer.page.metricsPanelTitle', {
-                            defaultMessage: 'Metrics',
-                          })}
-                          totalFieldCount={totalMetricFieldCount}
-                          populatedFieldCount={populatedMetricFieldCount}
-                          fieldTypes={[ML_JOB_FIELD_TYPES.NUMBER]}
-                          showFieldType={ML_JOB_FIELD_TYPES.NUMBER}
-                          showAllFields={showAllMetrics}
-                          setShowAllFields={setShowAllMetrics}
-                          fieldSearchBarQuery={metricFieldQuery}
-                          setFieldSearchBarQuery={setMetricFieldQuery}
-                          fieldVisConfigs={metricConfigs}
-                        />
-                        <EuiSpacer size="m" />
-                      </Fragment>
-                    )}
-                    <FieldsPanel
-                      title={i18n.translate('xpack.ml.datavisualizer.page.fieldsPanelTitle', {
-                        defaultMessage: 'Fields',
-                      })}
-                      totalFieldCount={totalNonMetricFieldCount}
-                      populatedFieldCount={populatedNonMetricFieldCount}
-                      showAllFields={showAllNonMetrics}
-                      setShowAllFields={setShowAllNonMetrics}
-                      fieldTypes={indexedFieldTypes}
-                      showFieldType={nonMetricShowFieldType}
-                      setShowFieldType={setNonMetricShowFieldType}
-                      fieldSearchBarQuery={nonMetricFieldQuery}
-                      setFieldSearchBarQuery={setNonMetricFieldQuery}
-                      fieldVisConfigs={nonMetricConfigs}
-                    />
-                  </EuiFlexItem>
-                </EuiFlexGroup>
+                <EuiPanel>
+                  <SearchPanel
+                    indexPattern={currentIndexPattern}
+                    searchString={searchString}
+                    setSearchString={setSearchString}
+                    searchQuery={searchQuery}
+                    setSearchQuery={setSearchQuery}
+                    searchQueryLanguage={searchQueryLanguage}
+                    samplerShardSize={samplerShardSize}
+                    setSamplerShardSize={setSamplerShardSize}
+                    totalCount={overallStats.totalCount}
+                  />
+                  <EuiHorizontalRule />
+                  <EuiFlexGroup gutterSize="m">
+                    <EuiFlexItem>
+                      {totalMetricFieldCount > 0 && (
+                        <Fragment>
+                          <FieldsPanel
```

Comment:
Think with both fields panels now appearing inside the same `EuiFlexItem` that a `key` should be set. Am seeing this error in the console:

![image](https://user-images.githubusercontent.com/7405507/72343524-ff210500-36c6-11ea-9c41-c256a1deae9e.png)

Reply:
https://github.com/elastic/kibana/pull/54358/commits/34ddc1f1b3ed76c4f0003e397730bdc9a0b03444